### DISCUSSION TITLE
Please, stop using location to set the website's language

### DISCUSSION THREAD

**Discussion by CamilleHbp** [solidjs/solid#1705] [Ideas]
My system and browser are set to English. I would like you to respect that instead of trying to guess my language base on location on [solidjs.com](https://www.solidjs.com/).

Would it be possible to first check the browser's language, and only if not set THEN use the location? I think this could be done in [this way](https://dev.to/bitdweller/stop-setting-the-language-of-your-website-based-on-my-location-31h0), unless I'm mising something?

It would be really nice to be able to stop switching back to my preferred language when visiting the website.

Thanks! 👍

**Comment by CamilleHbp** [comment]
What is the stance of the devs on this?

**Comment by atk** [comment]
User settings should always override any detection. Not sure what's wrong here.

**Reply by CamilleHbp** [reply]
Well, I'm using Brave, maybe the problem is on my side and some parameter prevents the correct detection of the setting by the frontend? If so, then sorry for raising a false alarm
